src/JsonOperationsTrait.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace JiraRestApi;
+
+use Exception;
+use JsonMapper;
+
+trait JsonOperationsTrait
+{
+    /**
+     * @param mixed $data
+     * @return string
+     * @throws Exception When cannot encode the given data (JsonException if PHP 7.3+).
+     */
+    protected function encodeJson($data): string
+    {
+        $ret = json_encode($data, $this->getJsonOptions());
```

Comment:
When relying on another method in the trait, it's important to have that method declared _in the trait_. Since for this trait the implementation of `getJsonOptions()` is unknowable, we cannot implement it here, but this just means it's abstract, and so we can declare `abstract protected function getJsonOptions()`. By doing so, we are formally declaring a dependency of the trait, while leaving the `use`r of the trait free to satisfy that dependency in any way they want, including by using another trait.

Reply:
:heavy_check_mark: